### DISCUSSION TITLE
I2C issues with LGT8F328P

### DISCUSSION THREAD

**Discussion by Jonnywayway** [dbuezas/lgt8fx#205] [Q&A] [answered]
Hello,

I'm working on a project where the LGT is  a worker to a Master ESP-01. I tested the code and the wiring with a Arduino Uno and I was able to receive/send messages. Using the same code and the same wiring I no longer am able to communicate to the ESP while using the LGT. Any ideas on what might be causing this and any ideas on how to fix it?

Thanks
Relevant code on the LGT

```
Wire.begin(i2cAddress); //i2c address of this unit
Wire.onReceive(receiveLetter); //call-function
Wire.onRequest(requestEvent);
```

```
void requestEvent() {
  Wire.write(currentlyrotating); //send unit status to master
  
    #ifdef serial
    Serial.print("Status ");
    Serial.print(currentlyrotating);
    Serial.print(" sent to master");
    Serial.println();
    #endif
}
```

```
 void receiveLetter(int numBytes) {
  int receiveArray[2]; //array for received bytes

  for (int i = 0; i < numBytes; i++) {
    receiveArray[i] = Wire.read();
  }
  //Write received bytes to correct variables
  receivedNumber = receiveArray[0];
  stepperSpeed = receiveArray[1];
}
```

**Comment by LaZsolt** [comment]
Did you choose the 16 MHz clock?

**Comment by L10N37** [comment]
I have issues with SPI. My PSXTAL project works fine (I'm using V1.00B atm) with 168/328P  nano clones but on this board, I lose my IGR (it uses SPI to capture packets). I hate these clone chips. I've tried quite a few on different projects and all have their quirks. 

The 168P clones won't work with my Uberchip modchip project, 328P does though.
168 works with PSXTAL
TTGO clone and LGT8F328P however, do not.

**Reply by LaZsolt** [reply]
The SPI works differently in the LGT8F328P than in the ATmega.
You have to use the SPI library from the dbuezas/lgt8fx package.

**Reply by L10N37** [reply]
Thanks so much!! scoured the web all night looking for new library to suit as I thought that was the case.. couldn't find squat. Next install I will look it up and make use of my little purple boards !! yay

**Comment by Jonnywayway** [comment, marked as answer]
So I did figure out what is going on, LGT8F328P will not respond to any of my commands if I set its I2C address below 8. I know the I2C standard reserves these for specific uses, but I would have assumed that the program would still respond to it even if its SUPPOSE to be used for a specific use case.

**Reply by LaZsolt** [reply]
So you answered your question. :)
The LGT's databook said the same address exceptions as the ATmegas's. The information you have discovered is new.

Edit: I mean reserved addresses.
